types/eslint-utils/index.d.ts

```diff
@@ -0,0 +1,284 @@
+
+/**
+ * Checks if the given token is an arrow token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is an arrow token.
+ */
+export function isArrowToken(node: CommentOrToken): node is AST.ArrowToken;
+
+/**
+ * Checks if the given token is a comma token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is a comma token.
+ */
+export function isCommaToken(node: CommentOrToken): node is AST.CommaToken;
+
+/**
+ * Checks if the given token is a semicolon token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is a semicolon token.
+ */
+export function isSemicolonToken(node: CommentOrToken): node is AST.SemicolonToken;
+
+/**
+ * Checks if the given token is a colon token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is a colon token.
+ */
+export function isColonToken(node: CommentOrToken): node is AST.ColonToken;
+
+/**
+ * Checks if the given token is an opening parenthesis token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is an opening parenthesis token.
+ */
+export function isOpeningParenToken(node: CommentOrToken): node is AST.OpeningParenToken;
+
+/**
+ * Checks if the given token is a closing parenthesis token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is a closing parenthesis token.
+ */
+export function isClosingParenToken(node: CommentOrToken): node is AST.ClosingParenToken;
+
+/**
+ * Checks if the given token is an opening square bracket token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is an opening square bracket token.
+ */
+export function isOpeningBracketToken(node: CommentOrToken): node is AST.OpeningBracketToken;
+
+/**
+ * Checks if the given token is a closing square bracket token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is a closing square bracket token.
+ */
+export function isClosingBracketToken(node: CommentOrToken): node is AST.ClosingBracketToken;
+
+/**
+ * Checks if the given token is an opening brace token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is an opening brace token.
+ */
+export function isOpeningBraceToken(node: CommentOrToken): node is AST.OpeningBraceToken;
+
+/**
+ * Checks if the given token is a closing brace token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is a closing brace token.
+ */
+export function isClosingBraceToken(node: CommentOrToken): node is AST.ClosingBraceToken;
+
+/**
+ * Checks if the given token is a comment token or not.
+ * @param node The comment or token to check.
+ * @returns `true` if the token is a comment token.
+ */
+export function isCommentToken(node: CommentOrToken): boolean;
```

Comment:
this would be nice as a type guard
```suggestion
export function isCommentToken(node: CommentOrToken): node is Comment;
```